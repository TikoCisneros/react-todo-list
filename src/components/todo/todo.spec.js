import React from 'react';
import Todo from './todo';
import renderer from 'react-test-renderer';

describe('Todo component', () => {
  it('renders correctly', () => {
    const props = {
      tasksData: [{ id: 1 }],
      filterOptions: [],
      onAddTask: jest.fn(),
      onSelectOption: jest.fn(),
      onClearCompleted: jest.fn(),
      onCheckTask: jest.fn(),
      onDeleteTask: jest.fn(),
    };
    const tree = renderer
      .create(<Todo {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
