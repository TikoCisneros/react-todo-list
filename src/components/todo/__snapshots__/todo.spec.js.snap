// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders correctly 1`] = `
Array [
  <header
    className="sc-fzozJi bpLWRU"
  >
    Simple Todo
  </header>,
  <div
    className="sc-AxmLO bSdOji"
  >
    <div
      className="sc-fzoLsD kVFGdJ"
    >
      <input
        className="sc-AxjAm bdELky"
        data-testid="form-input"
        onKeyPress={[Function]}
        placeholder="Write a task to be done!"
        type="input"
      />
      <div
        className="sc-AxirZ hlnYRa"
      >
        <div
          className="sc-AxiKw iQzYDT css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1wa3eu0-placeholder"
              >
                Chose a filter option...
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <button
          className="sc-AxhCb eprsCD"
          data-testid="clear-button"
          onClick={[MockFunction]}
        >
          Clear completed
        </button>
      </div>
    </div>
    <div
      className="sc-AxhUy cbSijq"
    >
      <input
        className="sc-AxgMl ibUEwm"
        data-testid="item-check"
        onChange={[Function]}
        type="checkbox"
      />
      <p
        className="sc-AxheI ImwVw"
        data-testid="item-name"
      />
      <button
        className="sc-Axmtr kVDdXX erase"
        data-testid="item-delete"
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>,
]
`;
